compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1404,12 +1406,15 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                         bounds,
                         fn_kind,
                         itctx,
-                        precise_capturing.as_deref().map(|(args, span)| (args.as_slice(), *span)),
                     ),
                     ImplTraitContext::Universal => {
-                        if let Some(&(_, span)) = precise_capturing.as_deref() {
+                        if let Some(span) = bounds.iter().find_map(|bound| match *bound {
+                            ast::GenericBound::Use(_, span) => Some(span),
+                            _ => None,
+                        }) {
                             self.tcx.dcx().emit_err(errors::NoPreciseCapturesOnApit { span });
```

Comment:
🤷 tho it's slightly inconsistent that we allow APITs with precise captures in `cfg(FALSE)` code

Reply:
Yeah, but we don't have a way of detecting whether something is an APIT or RPIT since the AST visitor doesn't keep track of whether we're in a universal or existential context. This is also preexisting.